apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/remote/GRPCChannelManager.java

```diff
@@ -146,16 +146,16 @@ public void run() {
                                                     .addChannelDecorator(new AgentIDDecorator())
                                                     .addChannelDecorator(new AuthenticationDecorator())
                                                     .build();
-                        notify(GRPCChannelStatus.CONNECTED);
                         reconnectCount = 0;
                         reconnect = false;
+                        notify(GRPCChannelStatus.CONNECTED);
                     } else if (managedChannel.isConnected(++reconnectCount > Config.Agent.FORCE_RECONNECTION_PERIOD)) {
                         // Reconnect to the same server is automatically done by GRPC,
                         // therefore we are responsible to check the connectivity and
                         // set the state and notify listeners
                         reconnectCount = 0;
-                        notify(GRPCChannelStatus.CONNECTED);
                         reconnect = false;
+                        notify(GRPCChannelStatus.CONNECTED);
```

Comment:
OK, the previous version had a change to cause a wrong status. 

Such as in `#notify`, the connection is disconnecting suddenly, then it would change the `reconnect=true`, but when notify finished, `reconnect` would be reset to false, which could cause no further connecting attempting.